### PR TITLE
Uploading Athena file and storing in postgres

### PR DESCRIPTION
**Pull request by kaelananderson**
# Overview

**Type of Change:**  New Feature

**Summary:**  I created a new method in upload_controller.js that effectively parsing an athena solar csv file that we can consistently expect from the user. The method creates a new endpoint in the routes called athena-upload. Using the code from the last work on the frontend, I integrated a user selecting athena as their choice of upload to be ingested into our postgres database. The file parses each of the time stamps at each site, and includes a summed total kWh for that timestamp, as well as a metadata created_at column. Unit testing included. 

## UI/UX Implementation

No changes made

### Data Models

New table: athena_hourly_output that holds timestamp for each record as well as kWh data for each site and a summed total as well.

### CODE REVIEW

File: backend/auth/upload/upload_controller.js [hunk 1]
```diff
@@ -109,4 +110,176 @@ export const uploadFile = async (req, res) => {
...
+            const allEmpty = siteHeaders.every((h) => {
+              const val = row[h];
+              return val === undefined || val === null || val.toString().trim() === '';
+            });
+            return !allEmpty;
+          });
+          console.log('Filtered Athena data (non-empty site columns):', parsedData);
+
+          // Transform each row to use the mapped column names and compute total_kwh.
+          const processedData = parsedData.map((row, rowIndex) => {
```
Comment (wyoung7788) [inline backend/auth/upload/upload_controller.js:216]: great capture of all the buildings that are included in the report and sets up database nicely to expect columns. test is nicely written to account for precision and any lacking in timestamps

File: frontend/src/components/UploadData.vue [hunk 2]
```diff
@@ -70,13 +69,29 @@
...
 const selectedFile = ref(null);
-const uploadedFileName = ref(''); // State to store the uploaded file name
+const uploadedFileName = ref(''); // Initialize as empty string
+const selectedSource = ref(localStorage.getItem("selectedSource") || "Athena");
+
+// Computed property for accepted file type based on selected source
+const acceptedFileType = computed(() => {
+  return selectedSource.value === 'Athena' ? '.csv' : '.xlsx';
+});
+
```
Comment (wyoung7788) [inline frontend/src/components/UploadData.vue:85]: great check for validating the file upload before processing the data

File: backend/auth/upload/upload_controller.js [hunk 3]
```diff
@@ -109,4 +110,176 @@ export const uploadFile = async (req, res) => {
...
+            const processedRow = {};
+            let totalKwh = 0;
+
+            // Helper function to normalize numbers to fixed precision
```
Comment (wyoung7788) [inline backend/auth/upload/upload_controller.js:220]: clear commenting on the functionality, helps the team to understand the preprocessing and how the database is being set up

File: frontend/src/components/UploadData.vue [hunk 4]
```diff
@@ -87,15 +102,17 @@ const handleUpload = async (_event) => {
...
-  for (let pair of formData.entries()) {
-    console.log(pair[0] + ': ' + pair[1].name);
-  }
-
   try {
     console.log('Making POST request to upload file...');
-    console.log('FormData:', formData);
-
-    const response = await axios.post('http://localhost:3000/api/auth/file-upload', formData, {
+
```
Comment (wyoung7788) [inline frontend/src/components/UploadData.vue:107]: good job setting up the api endpoint to store new data! it follows the setup we have for bloom and this makes it easy for us to adapt to different energy sources

File: backend/auth/upload/upload_controller.js [hunk 5]
```diff
@@ -109,4 +110,176 @@ export const uploadFile = async (req, res) => {
...
+            'University of San Diego - Jenny Craig Pavillion': 'jenny_craig_pavilion',
+            'University of San Diego - Kroc': 'kroc',
+            'University of San Diego - Manchester A': 'manchester_a',
+            'University of San Diego - Manchester B': 'manchester_b',
+            'University of San Diego - Soles': 'soles',
+            'University of San Diego - West Parking': 'west_parking'
+          };
 
+          // Map each data row (array) into an object using the header row as keys.
+          let parsedData = dataRows.map((row) => {
```
Comment (wyoung7788) [inline backend/auth/upload/upload_controller.js:185]: nice work on catching any possible edge cases in the spreadsheet and creating error messages so that we will be able to catch and identify any problems quickly

**Review by wyoung7788 — APPROVED** [review]
awesome job writing the tests that fully capture and test the functionality of the athena file upload as well as writing clear comments

**Comment by kaelananderson** [comment]
[Using Athena files, generate the schema in the database](https://trello.com/c/XRq8Mgff/60-using-athena-files-generate-the-schema-in-the-database)

**Comment by kaelananderson** [comment]
[Unit testing for file upload for Athena](https://trello.com/c/wVYX5Wr6/63-unit-testing-for-file-upload-for-athena)

**Comment by kaelananderson** [comment]
[Add functionality for uploading data from Athena file to database](https://trello.com/c/K6Vzu0g5/62-add-functionality-for-uploading-data-from-athena-file-to-database)
